src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryScreen.java

```diff
@@ -43,4 +61,103 @@ public EntityRef get() {
     public boolean isModal() {
         return false;
     }
+
+    /*
+      The numbersBetween() and getTransferEntity() methods were
+      originally in the InventoryCell class. They were copied over
+      to here because they are private functions that the
+      moveItemSmartly() method needs to function. The first section
+      of code in onClosed() is based on the moveItemSmartly() method.
+    */
+    private List<Integer> numbersBetween(int start, int exclusiveEnd) {
```

Comment:
Maybe we should make this package private if we need it in several places and don't inline it.